dashboard/app/reporting.go

```diff
@@ -1056,6 +1056,38 @@ func lastReportedReporting(bug *Bug) *BugReporting {
 	return nil
 }
 
+// The updateReporting method is supposed to be called both to fully initialize a new
+// Bug object and also to adjust it to the updated namespace configuration.
+func (bug *Bug) updateReportings(cfg *Config, now time.Time) {
+	oldReportings := map[string]BugReporting{}
+	for _, rep := range bug.Reporting {
+		oldReportings[rep.Name] = rep
+	}
+	bug.Reporting = []BugReporting{}
+	for _, rep := range cfg.Reporting {
+		if oldRep, ok := oldReportings[rep.Name]; ok {
+			bug.Reporting = append(bug.Reporting, oldRep)
+		} else {
+			bug.Reporting = append(bug.Reporting, BugReporting{
+				Name: rep.Name,
+				ID:   bugReportingHash(bug.keyHash(), rep.Name),
+			})
+		}
+	}
+	// We might have added new BugReporting objects between/before the ones that were
+	// already reported. To let syzbot continue from the same reporting stage where it
+	// stopped, close such outliers.
+	seenProcessed := false
+	for i := len(bug.Reporting) - 1; i >= 0; i-- {
+		rep := &bug.Reporting[i]
+		if !rep.Reported.IsZero() || !rep.Closed.IsZero() || !rep.OnHold.IsZero() {
+			seenProcessed = true
+		} else if seenProcessed {
+			rep.Closed = now
```

Comment:
> We could also create a special field in BugReporting

Check what we do when a stage is skipped due to config filter (ReportSkip). I think it would be reasonable to do the same rather than add a rarely exercised special case.